### PR TITLE
Adding a configuration field to TERCC.

### PR DESCRIPTION
**Pull request by t-persson**
Not sure if we add new json schemas in PR like this, so i added one. Please inform me if that should be removed.

### Applicable Issues

fixes: #241 

### Description of the Change

This configuration field will be used for batch-level configuration
of a test suite. This is not to be confused with the constraints field
which is for a recipe-level configuration.

### Alternate Designs

Today we are using 'data.cusomData' for the same feature, but customData should be avoided.

### Benefits

Avoiding the usage of the customData field is always a benefit.

### Possible Drawbacks

Not sure if this field will be usable by others. Might add unnecessary clutter.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobiaspn@axis.com>

### CODE REVIEW

**Comment by e-backmark-ericsson** [comment]
The updated JSON schema should be included, so you did it right in adding it :) I'll review this PR in a few days hopefully.

File: eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md [hunk 1]
```diff
@@ -67,6 +67,21 @@ __Type:__ Integer
 __Required:__ Yes  
 __Description:__ The execution priority of the batch, as compared to other batches in the collection. A lower value SHALL be interpreted as a higher priority.
 
+#### data.batches.configuration
```
Comment (e-backmark-ericsson) [inline eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md:70]: As this is intended to keep the same kind of information as data.batches.recipes.constraints, I'd like to see this parameter called data.batches.constraints instead. If you don't think that is suitable you should instead suggest to change the name of the existing parameter (or rather make the current parameter deprecated and propose the new).

File: eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md [hunk 2]
```diff
@@ -67,6 +67,21 @@ __Type:__ Integer
...
+__Type:__ Object[]
+__Required:__ No
+__Description:__ Any configuration required for all recipes. This is a custom field and is highly dependent on the test execution framework. Typically used for the set up of the execution space where the test cases can be executed. Example could be to override default configurations in the test execution framework. This property is a list of key-value pairs in the same format as [data.customData](../customization/custom-data.md). Note that this is a per-batch configuration and will apply to all recipes. Not to be confused with __data.batches.recipes.constraints__ which apply to only a single recipe.
```
Comment (e-backmark-ericsson) [inline eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md:73]: Again, as this parameter is the same as recipe constraints, the description should be more similar to that one. Also, please state the need to be careful if providing both batch level constraints and recipe level constraints to the same TERCC. I'd like to see that elaborate on the top of this page and then referenced from here as well as from the documentation for the recipe constraints.

File: schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json [hunk 3]
```diff
@@ -0,0 +1,295 @@
+{
```
Comment (e-backmark-ericsson) [inline schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json:1]: This should probably be 4.1.0 instead according to the versioning description here: https://github.com/eiffel-community/eiffel/blob/master/eiffel-syntax-and-usage/versioning.md

File: schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+          "type": "string",
+          "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+        },
+        "type": {
+          "type": "string",
+          "enum": [ "EiffelTestExecutionRecipeCollectionCreatedEvent" ]
+        },
+        "version": {
+          "type": "string",
+          "enum": [ "4.0.0" ],
```
Comment (e-backmark-ericsson) [inline schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json:18]: 4.1.0

File: schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+          "default": "4.0.0"
```
Comment (e-backmark-ericsson) [inline schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json:19]: 4.1.0

File: schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string"
+              },
+              "priority": {
+                "type": "integer"
+              },
+	      "configuration": {
```
Comment (e-backmark-ericsson) [inline schemas/EiffelTestExecutionRecipeCollectionCreatedEvent/5.0.0.json:147]: constraints
(see comment in .md file)

File: eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md (hunk 1 above)
Comment (t-persson) [reply eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md]: Thanks for the comment. Renamed the parameter.

File: eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md (hunk 2 above)
Comment (t-persson) [reply eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md]: Changed documentation, please evaluate

**Pull request closed without merging**